lib/compiler/src/engine/resolver.rs

```diff
@@ -104,14 +104,17 @@ pub fn resolve_imports(
         }
         match *resolved {
             VMExtern::Function(handle) => {
-                let f = handle.get(context);
+                let f = handle.get_mut(context);
                 let address = match f.kind {
                     VMFunctionKind::Dynamic => {
                         // If this is a dynamic imported function,
                         // the address of the function is the address of the
                         // reverse trampoline.
                         let index = FunctionIndex::new(function_imports.len());
-                        finished_dynamic_function_trampolines[index].0 as *mut VMFunctionBody as _
+                        let ptr = finished_dynamic_function_trampolines[index].0
```

Comment:
This seems a somewhat hacky place to put this code, but if it is hard to put somewhere else, there shoud be a comment explaining why this is here.

Reply:
We can put this comment:

```
/// TODO (@syrusakbary): We need to do this properly, handling this well in the VM while getting rid of `VMFunctionKind`
```

Reply:
And a bit more explanation as well. ;)

Reply:
I'll let Syrus come up with the comment, as I don't quite understand the situation well enough.

Reply:
@syrusakbary LMK what you want me to put in there, so I can add commits in both places (this and the test) and we can merge this in afterwards.

Reply:
```
/// The logic is currently handling the "resolution" of dynamic imported functions at instantiation time.
/// However, ideally it should be done even before then, as you may have dynamic imported functions that are linked at runtime and not instantiation time. And those will not work properly with the current logic.
/// Ideally, this logic should be done directly in the `wasmer-vm` crate.
/// TODO (@syrusakbary): Get rid of `VMFunctionKind`
```